src/index.ts

```diff
@@ -176,11 +176,11 @@ type EveryKeyPresent<A, B> = Array<A> extends B
  */
 export const getCodecFromMappedNullaryTag =
   <A extends NullaryMember>(sum: Sum.Sum<A>) =>
-  <B>(from: (x: unknown) => O.Option<Tag<A>>, to: (x: Tag<A>) => B) =>
+  <O, I>(from: (x: I) => O.Option<Tag<A>>, to: (x: Tag<A>) => O) =>
```

Comment:
```suggestion
  <O, I = unknown>(from: (x: I) => O.Option<Tag<A>>, to: (x: Tag<A>) => O) =>
```

I think we need to set `I` as default to `unknown` else this is a breaking change.

Reply:
If it's inferring where previously it couldn't one could argue this is a bugfix.

Reply:
> If it's inferring where previously it couldn't one could argue this is a bugfix.

@samhh What would you do in this case then?

On a related note, I wonder if we should refrain from specifying these generics explicitly and instead use type annotations. For example:

```diff
 const WeatherFromCountry: t.Type<Weather, Country> =
-  getCodecFromMappedNullaryTag(Weather)<Country, unknown>(
-    x => {
+  getCodecFromMappedNullaryTag(Weather)(
+    (x: unknown) => {
       switch (x) {
         case "Italy":
           return O.some("Sun")
         case "UK":
           return O.some("Rain")
         default:
           return O.none
       }
     },
-    x => (x === "Sun" ? "Italy" : "UK"),
+    (x): Country => (x === "Sun" ? "Italy" : "UK"),
   )(["Sun", "Rain"])
```

Reply:
I'd be tempted to make the change and release as a patch. `@types` packages have similar difficulties qualifying what is and isn't breaking. Perhaps first patch the types in Web and run the typechecker to see how much damage it can cause.